src/stores/reactions.js

```diff
@@ -0,0 +1,218 @@
+		},
+
+		/**
+		 * Adds reactions for a given message.
+		 *
+		 * @param {object} payload action payload
+		 * @param {string} payload.token The conversation token
+		 * @param {number} payload.messageId The id of message
+		 * @param {object} payload.reactions The list of reactions with details for a given message
+		 *
+		 */
+		addReactions({ token, messageId, reactions }) {
+			if (!this.reactions[token]) {
+				Vue.set(this.reactions, token, {})
+
+			}
+			Vue.set(this.reactions[token], messageId, reactions)
+		},
+
+		/**
+		 * Delete all reactions for a given message.
+		 *
+		 * @param {string} token The conversation token
+		 * @param {number} messageId The id of message
+		 *
+		 */
+		resetReactions(token, messageId) {
+			if (!this.reactions[token]) {
+				Vue.set(this.reactions, token, {})
+			}
+			Vue.delete(this.reactions[token], messageId)
+		},
+
+		/**
+		 * Updates reactions for a given message.
+		 *
+		 * @param {object} payload action payload
+		 * @param {string} payload.token The conversation token
+		 * @param {number} payload.messageId The id of message
+		 * @param {object} payload.reactionsDetails The list of reactions with details for a given message
+		 *
+		 */
+		updateReactions({ token, messageId, reactionsDetails }) {
+			if (!this.reactions[token]) {
+				Vue.set(this.reactions, token, {})
+			}
+
+			if (!this.reactions[token][messageId]) {
+				Vue.set(this.reactions[token], messageId, [])
+			}
+
+			const storedReactions = this.reactions[token][messageId]
+
+			if (Object.keys(reactionsDetails).length === 0) {
+				this.resetReactions(token, messageId)
+				return
+			}
+
+			if (Object.keys(storedReactions).length === 0) {
+				Vue.set(this.reactions[token], messageId, reactionsDetails)
+				return
+			}
+
+			// Handle removed reactions
+			const removedReactions = Object.keys(storedReactions).filter(reaction => {
+				return !reactionsDetails[reaction]
+			})
+			if (Object.keys(removedReactions).length > 0) {
+				removedReactions.forEach(reaction => {
+					this.removeReaction({ token, messageId, reaction })
+				})
+			}
+
+			// Add new reactions and update existing ones
+			Object.entries(reactionsDetails).forEach(([reaction, actors]) => {
+				if (!storedReactions[reaction]) {
+					// Reaction does not exsist, so add it
+					this.addReaction({ token, messageId, reaction, actors })
+				} else if (JSON.stringify(actors) !== JSON.stringify(storedReactions[reaction])) {
+					// Reaction does exsist, so check update actors
+					this.addReaction({ token, messageId, reaction, actors })
+				}
+			})
+		},
+
+		/**
+		 * Gets the full reactions list for a given message.
+		 *
+		 * @param {string} token The conversation token
+		 * @param {number} messageId The id of message
+		 *
+		 */
+		async fetchReactions(token, messageId) {
+			console.debug('getting reactions details')
+			try {
+				const response = await getReactionsDetails(token, messageId)
+				this.updateReactions({
```

Comment:
Here we could patch messagesStore with new plain reactions data, if they are outdated. And in that case next fetch won't trigger a detailed reactions reset

Reply:
I don't really understand how you see the patch is resetting the store 👾. Reset only happens when no reactions left after removing all emojis ( that means the data from the server comes empty and thus reset is rare to happen)